### PR TITLE
Use jax.random for stax initialization

### PR DESCRIPTION
**Pull request by j-towns**
This is a stax API breaking change (not sure how open people are to those).

This fixes part of https://github.com/google/jax/issues/245, because weights will now be initialized on device. It also allows one to easily have a single PRNG seed for a whole experiment, rather than having to manage separate Numpy and JAX PRNGs.

### CODE REVIEW

**Comment by mattjj** [comment]
We are open to breaking the stax and optimizers APIs, hence why they're in `jax.experimental`. More importantly, this is a feature that several others have asked for (including @gdahl and more recently others in our chat room), so this is as good a reason as we can get for API tweaks.

Thanks for so thoroughly updating all the example code!

**Comment by j-towns** [comment]
The only thing I didn't update is this note: https://github.com/google/jax/blob/master/design_notes/prng.md, which describes a few stax layers in the old API, happy to update it if it's helpful to do so.

**Review by mattjj — APPROVED** [review]
Fantastic work as always.

In the diff it looks like the maml notebook might be corrupted for uninteresting reasons. I'm going to try to fix that in a follow-up.
